interfaces/builtin/unity8_pim_common.go

```diff
@@ -108,34 +109,35 @@ func (iface *unity8PimCommonInterface) PermanentPlugSnippet(plug *interfaces.Plu
 	return nil, nil
 }
 
-func (iface *unity8PimCommonInterface) ConnectedPlugSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
-	switch securitySystem {
-	case interfaces.SecurityAppArmor:
-		old := []byte("###SLOT_SECURITY_TAGS###")
-		new := slotAppLabelExpr(slot)
+func (iface *unity8PimCommonInterface) AppArmorConnectedPlug(spec *apparmor.Specification, plug *interfaces.Plug, slot *interfaces.Slot) error {
+	old := "###SLOT_SECURITY_TAGS###"
+	new := slotAppLabelExpr(slot)
 
-		originalSnippet := append([]byte(unity8PimCommonConnectedPlugAppArmor), iface.connectedPlugAppArmor...)
-		snippet := bytes.Replace(originalSnippet, old, new, -1)
+	originalSnippet := unity8PimCommonConnectedPlugAppArmor + "\n" + iface.connectedPlugAppArmor
+	snippet := strings.Replace(originalSnippet, old, new, -1)
 
-		// classic mode
-		if release.OnClassic {
-			// Let confined apps access unconfined service on classic
-			classicSnippet := bytes.Replace(originalSnippet, old, []byte("unconfined"), -1)
-			snippet = append(snippet, classicSnippet...)
-		}
-
-		return snippet, nil
-	default:
-		return nil, nil
+	// classic mode
+	if release.OnClassic {
+		// Let confined apps access unconfined service on classic
+		classicSnippet := strings.Replace(originalSnippet, old, "unconfined", -1)
```

Comment:
This interface is pretty awkward. We need to confirm with @jdstrand that this is really what we want here.

But no need to block on this. It was already like that before.